bin/endpoint.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A script for manipulating the endpoint instance.
+
+COMMANDS : A dictionary of available commands and the functions to call
+"""
+
+import argparse
+import boto3
+from collections import namedtuple
+from lib import aws
+from lib.external import ExternalCalls
+from lib.names import AWSNames
+from mysql import connector
+
+DB_HOST_NAME = 'endpoint-db'
+
+class ResourceNotFoundException(Exception):
+    """
+    Raised when unable to locate the id of collection, experiment, or 
+    resource.
+    """
+
+"""
+Container for MySQL connection parameters.
+
+Fields:
+    host (str): DB host name or ip address.
+    port (str|int): Port to connect to.
+    db (str): Name of DB.
+    user (str): DB user name.
+    password (str): User password.
+"""
+DbParams = namedtuple('DbParams', ['host', 'port', 'db', 'user', 'password'])
+
+"""
+Container that identifies Boss channel.
+
+Fields:
+    collection (str): Collection name.
+    experiment (str): Experiment name.
+    channel (str): Channel name.
+"""
+ChannelParams = namedtuple(
+    'ChannelParams', ['collection', 'experiment', 'channel'])
+
+def get_mysql_params(session, domain):
+    """
+    Get MySQL connection info from Vault.
+
+    Args:
+        session (Session): Open boto3 session.
+        domain (str): VPC such as integration.boss.
+
+    Returns:
+        (DbParams): Connection info from Vault.
+    """
+    keypair = aws.keypair_lookup(session)
+    call = ExternalCalls(session, keypair, domain)
+    names = AWSNames(domain)
+
+    names = AWSNames(domain)
+    print('Getting MySQL parameters from Vault (slow) . . .')
```

Comment:
While I understand the intent with the print statements, the problem they pose is that if someone wants to use this script in a pipeline or shell script it will pollute the output and harder to use.

Maybe a `-q` quiet flag, or similar, to allow modification of the output without removing it completely?

Reply:
Yeah this sounds good.